internal/state/store.go

```diff
@@ -420,6 +427,14 @@ func (store dbStore) PruneStates(from int64, to int64, evidenceThresholdHeight i
 		return err
 	}
 
+	// We do not want to panic or interrupt consensus on compaction failure
+	if store.StoreOptions.Compact && (to%store.StoreOptions.CompactionInterval == 0 || pruned >= uint64(store.StoreOptions.CompactionInterval)) {
```

Comment:
I'm trying to understand this second part of the `if`. In `to%store.StoreOptions.CompactionInterval == 0` is this a good condition ? I'm thinking that let's say `to = 100` but the `CompactionInterval = 10` then a compaction would happen, the question is, does the `to` increases sequently or it could skip (e.g. 100, 105, etc) ? If the latter then the compaction might not happen in the desired interval.

Reply:
But `pruned` is never reset to 0

Reply:
@andynog  Yes you are right, if `to` is not a multiple of `CompactionInterval` then we have a problem. I will see how to change it. 

> But pruned is never reset to 0

It is set at the beginning of the function every time we call it.

Reply:
Changed it but had to adapt the interface of the PruneStates function .